Raven.NET.Core/Providers/RavenProvider.cs

```diff
@@ -9,16 +10,30 @@ namespace Raven.NET.Core.Providers
     public class RavenProvider : IRavenProvider
     {
         /// <inheritdoc/>
-        public bool AddRaven(string ravenName, IRaven raven) => RavenCache.RavenWatcherCache.TryAdd(ravenName, raven);
+        public bool AddRaven(string ravenName, IRaven raven, Type subjectType = default)
```

Comment:
what do you think here about switch expression? easier to extend it in future ( just opinion )

Reply:
Love them honestly, but for this case it does not make a lot of sense using it, because i cannot imagine another type of Raven. But If there will be third one then its fair to change it for switch expression. If you only have 2 options then i believe its even better to just use if condition, its just clearer for the case.